### DISCUSSION TITLE
How to kill a background targets pipeline created with the "Run a targets pipeline in the background" Addin in Workbench?

### DISCUSSION THREAD

**Discussion by sctyner** [ropensci/targets#1011] [Help]
Hello,

I have a pipeline that appears to have been running in the background for over 4 hours. I created it with the ["Run a targets pipeline in the background" Addin](https://github.com/ropensci/targets/blob/7a76cf8c94145f2f4b70f0859dba7eb5d918396c/R/rstudio_addin_tar_make_bg.R). I want to kill it. I cannot figure out how to. 

I have rebooted everything twice and it is still running. Note: I am on my company's professional Posit Worbench Pro Server. By "reboot" I mean I have force quit all of my open sessions and (allegedly) all of their child processes (see below screen grab). 

![Screenshot 2023-01-13 at 8 03 03 PM](https://user-images.githubusercontent.com/6137322/212443258-9fa44e9e-17e5-44f0-ae73-e1abd5c84348.png)

However, when I open up the same Rstudio project in a brand new session, the background session still appears to be running (`targets::tar_process_summary()` since column value is increasing, see below screen grab). And I am confident it is still taking up computing resources because computations are running much slower than they were before I ran the Addin. 

![Screenshot 2023-01-13 at 8 04 22 PM](https://user-images.githubusercontent.com/6137322/212443257-3c449d9a-cdf7-4066-a5a5-eafd8cd4df9c.png)

Please help me kill this background process! 

Thank you,
Sam

**Comment by wlandau** [comment]
tar_process() returns the process ID of the most recent run of the pipeline, and if your Workbench session is logged into the node where that process is running, you can terminate it with ps::ps_kill(). If your company’s installation of Workbench uses multiple nodes/host machines, then it may be tricky to find the one with your process. (For privacy reasons, targets does not record the host name of the process.) It might help to try different nodes and see which one shows the process when you run ps::ps().

**Reply by wlandau** [reply]
In the future, you could run something like px <- targets::tar_make(callr_function = callr::r_bg). This is what the addin does, except you know have a callr handle object px that helps you terminate the process if needed.

**Comment by sctyner** [comment]
Thanks for the response. Unfortunately I was not able to kill it with `ps_kill()`.

![Screenshot 2023-01-17 at 9 49 38 AM](https://user-images.githubusercontent.com/6137322/212930259-e0be5880-d631-4f98-8126-370d3464de56.png)

Could you provide more detail on how to "try different nodes"?

**Reply by wlandau** [reply]
If your company's infrastructure is like mine, then your installation of Workbench may be backed by multiple hosts. That means `Sys.info()["nodename"]` could be different every time you log in and/or restart your R session. That process may be difficult to find and terminate because you might need to be logged into the host that originally launched it. I recommend reaching out to your sys admin to try and track down the process. It may be useful to useful to have a list of all the node names running Workbench so you can SSH into each one to look for stale R sessions.
